lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -78,18 +106,27 @@ export default function SubmitGradeForm({ disabled = false, student }) {
       <label className="SubmitGradeForm__label" htmlFor="lms-grade">
         Grade (Out of 10)
       </label>
-      <input
-        className={classNames('SubmitGradeForm__grade', {
-          'SubmitGradeForm__grade--saved': requestStatus === 'done',
-        })}
-        disabled={disabled}
-        id="lms-grade"
-        ref={refInput}
-        onKeyDown={handleKeyDown}
-        type="input"
-        defaultValue={''} // TODO: stub until we pull the grade from the service
-        key={student.LISResultSourcedId}
-      />
+      <span className="SubmitGradeForm__grade-wrapper">
+        <input
+          className={classNames('SubmitGradeForm__grade', {
+            'SubmitGradeForm__grade--saved': requestStatus === 'done',
+          })}
+          disabled={disabled}
+          id="lms-grade"
+          ref={refInput}
+          onKeyDown={handleKeyDown}
+          type="input"
+          defaultValue={grade}
+          key={grade}
```

Comment:
Can you explain why the `key` for this element is the current value of the `grade`? (And why this element needs a `key` at all?)

Reply:
It's really just to reset the field when the grade changes to the value updates, but it perhaps makes more sense to base this off the student. It has the same effect either way, but I changed this to the student id so its more clear.

If we don't have a key at all then the grade will never change because `defaultValue` only works once when the component is created and each subsequent change will not change the value.